src/contracts/rewardsV2/FeeRegistry.sol

```diff
@@ -2,320 +2,207 @@
-        (bool isEnabled, uint256 feeValue) = getOperatorVaultNetworkFee(operator, vault, network);
-        if (isEnabled) return feeValue;
-
-        (isEnabled, feeValue) = getOperatorNetworkFee(operator, network);
-        if (isEnabled) return feeValue;
+    /* CONSTRUCTOR */
 
-        (isEnabled, feeValue) = getOperatorVaultFee(operator, vault);
-        if (isEnabled) return feeValue;
-
-        (isEnabled, feeValue) = getOperatorGlobalFee(operator);
-        if (isEnabled) return feeValue;
-
-        return DEFAULT_OPERATOR_FEE;
+    constructor(
+        address curatorRegistry_
+    ) {
+        if (curatorRegistry_ == address(0)) {
+            revert CuratorRegistryIsZero();
+        }
+        curatorRegistry = ICuratorRegistry(curatorRegistry_);
     }
 
-    /**
-     * @inheritdoc IFeeRegistry
-     */
-    function getOperatorGlobalFeeAt(
-        address operator,
-        uint48 timestamp,
-        bytes memory hint
-    ) public view returns (bool, uint256) {
-        return _deserializeFeeData(_operatorGlobalFee[operator].upperLookupRecent(timestamp, hint));
-    }
+    /* FUNCTIONS */
 
     /**
      * @inheritdoc IFeeRegistry
      */
-    function getOperatorGlobalFee(
-        address operator
-    ) public view returns (bool, uint256) {
-        return _deserializeFeeData(_operatorGlobalFee[operator].latest());
-    }
+    function getOperatorFeeAt(address vault, address network, uint48 timestamp) public view returns (uint256 fee) {
+        // Check if network-specific fee exists and is enabled
+        (bool isEnabled, uint256 networkFee) = getOperatorNetworkFeeAt(vault, network, timestamp);
+        if (isEnabled) {
+            return networkFee;
+        }
 
-    /**
-     * @inheritdoc IFeeRegistry
-     */
-    function getOperatorVaultFeeAt(
-        address operator,
-        address vault,
-        uint48 timestamp,
-        bytes memory hint
-    ) public view returns (bool, uint256) {
-        return _deserializeFeeData(_operatorVaultFee[operator][vault].upperLookupRecent(timestamp, hint));
+        // Fall back to global vault fee
+        return getOperatorFee(vault, network);
     }
 
     /**
      * @inheritdoc IFeeRegistry
      */
-    function getOperatorVaultFee(address operator, address vault) public view returns (bool, uint256) {
-        return _deserializeFeeData(_operatorVaultFee[operator][vault].latest());
+    function getOperatorFee(address vault, address network) public view returns (uint256 fee) {
+        // Check if network-specific fee exists and is enabled
+        (bool isEnabled, uint256 networkFee) = getOperatorNetworkFee(vault, network);
+        if (isEnabled) {
+            return networkFee;
+        }
+
+        // Fall back to global vault fee
+        return _operatorFee[vault].latest();
     }
 
     /**
      * @inheritdoc IFeeRegistry
      */
     function getOperatorNetworkFeeAt(
```

Comment:
In this case, let's leave a named return, as there can be less clarity why are there 2 variables